db/sqldb/populate.go

```diff
@@ -0,0 +1,158 @@
+package sqldb
+
+import (
+	"fmt"
+
+	"github.com/RMI/pacta/db"
+	"github.com/RMI/pacta/pacta"
+)
+
+type HasBlobs interface {
+	Blobs() []*pacta.Blob
+}
+
+func (d *DB) PopulateBlobs(tx db.Tx, hbs []HasBlobs) error {
```

Comment:
I don't think the database layer is the right location for this type of pointer twiddling, the telltale sign is that it doesn't actually do anything specific to the Postgres DB implementation. It could comfortably live at the server layer.

Also, do we need this whole paradigm here? I can understand it in a GraphQL server where you have arbitrarily nested things, but in a more REST-like server, you always know exactly what data you're returning and can just populate it in one fell swoop, there's no dynamicness

Reply:
Very good point - I'll remove this.